src/garage/envs/multi_env_wrapper.py

```diff
@@ -150,96 +182,104 @@ def active_task_index(self):
 
-        return self._active_task_index
-
-    @property
-    def observation_space(self):
-        """Observation space.
-
-        Returns:
-            akro.Box: Observation space.
-
-        """
-        if self._mode == 'vanilla':
-            return self.env.observation_space
-        elif self._mode == 'add-onehot':
-            task_lb, task_ub = self.task_space.bounds
-            env_lb, env_ub = self._observation_space.bounds
-            return akro.Box(np.concatenate([env_lb, task_lb]),
-                            np.concatenate([env_ub, task_ub]))
-        else:  # self._mode == 'del-onehot'
-            env_lb, env_ub = self._observation_space.bounds
-            num_tasks = self._num_tasks
-            return akro.Box(env_lb[:-num_tasks], env_ub[:-num_tasks])
-
-    @observation_space.setter
-    def observation_space(self, observation_space):
-        """Observation space setter.
-
-        Args:
-            observation_space (akro.Box): Observation space.
-
-        """
-        self._observation_space = observation_space
-
-    def _active_task_one_hot(self):
-        """One-hot representation of active task.
-
-        Returns:
-            numpy.ndarray: one-hot representation of active task
-
-        """
-        one_hot = np.zeros(self.task_space.shape)
-        index = self.active_task_index or 0
-        one_hot[index] = self.task_space.high[index]
-        return one_hot
-
     def reset(self, **kwargs):
         """Sample new task and call reset on new task env.
 
         Args:
-            kwargs (dict): Keyword arguments to be passed to gym.Env.reset
+            kwargs (dict): Keyword arguments to be passed to
+            Environment.reset()
 
         Returns:
-            numpy.ndarray: active task one-hot representation + observation
+            numpy.ndarray: The first observation. It must conforms to
+            `observation_space`.
+            dict: The episode-level information. Note that this is not part
+            of `env_info` provided in `step()`. It contains information of
+            the entire episode， which could be needed to determine the first
+            action (e.g. in the case of goal-conditioned or MTRL.)
 
         """
         self._active_task_index = self._sample_strategy(
             self._num_tasks, self._active_task_index)
         self.env = self._task_envs[self._active_task_index]
-        obs = self.env.reset(**kwargs)
+        obs, episode_info = self.env.reset(**kwargs)
+        self._step_cnt = 0
         if self._mode == 'vanilla':
-            return obs
+            pass
         elif self._mode == 'add-onehot':
-            return np.concatenate([obs, self._active_task_one_hot()])
+            obs = np.concatenate([obs, self._active_task_one_hot()])
         else:  # self._mode == 'del-onehot'
-            return obs[:-self._num_tasks]
+            obs = obs[:-self._num_tasks]
+
+        self._last_observation = obs
+
+        return obs, episode_info
```

Comment:
you can put the `active_task_one_hot` into episode_info! (just use the key `task`)

Reply:
Create a new PR for this? Changing this will affect how downstream consume the obs and this PR is already huge.

Reply:
okay, please leave an issue